### PR TITLE
Rework bin codacycoverage to use symfony/console

### PR DESCRIPTION
**Pull request by kirillkomlev**
- rework by use symfony/console
- add support git-commit option
- add support base-url option
- small fixes in CodacyApiClient

### CODE REVIEW

File: src/Codacy/Coverage/Command/Clover.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+ * Class Clover
+ *
+ */
+class Clover extends ConsoleCommand {
+
+    protected function configure()
+    {
+        $this
+            ->setName("clover")
+            ->setDescription("Send coverage results from clover.xml format")
```
Comment (haffla) [inline src/Codacy/Coverage/Command/Clover.php]: Maybe better: Send coverage results in clover format

File: src/Codacy/Coverage/Command/Clover.php [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            ->addArgument(
+                "path_to_coverage_results",
+                InputArgument::OPTIONAL,
+                "Path where coverage results saved"
```
Comment (haffla) [inline src/Codacy/Coverage/Command/Clover.php]: *Path where coverage results are saved: XML file for clover format; directory containing the index.xml for phpunit format.

File: src/Codacy/Coverage/Command/Clover.php [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            )
+            ->addOption(
```
Comment (haffla) [inline src/Codacy/Coverage/Command/Clover.php]: I am trying to make sense of this. Why should anyone want to specify the commit?

File: src/Codacy/Coverage/Command/Clover.php [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        $commit = $this->getCommitHash($input->getOption("git-commit"));
+
+        $baseUrl = $this->getBaseCodacyUrl($input->getOption("base-url"));
+
+        $url = $baseUrl . "/api/coverage/" . $projectToken . "/" . $commit;
+
+        $data = $jsonProducer->makeJson();
+
+        if ($output->isVerbose()) {
+            $output->writeln("Send coverage result to ". $url);
```
Comment (haffla) [inline src/Codacy/Coverage/Command/Clover.php]: *Sending

File: src/Codacy/Coverage/Command/Phpunit.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ */
+class Phpunit extends Clover {
+
+    protected function configure()
+    {
+        parent::configure();
+
+        $this
+            ->setName("phpunit")
+            ->setDescription("Send coverage results from phpunit coverage-xml dir");
```
Comment (haffla) [inline src/Codacy/Coverage/Command/Phpunit.php]: *Send coverage results in phpunit format

File: src/Codacy/Coverage/Command/Clover.php (hunk 3 above)
Comment (kirillkomlev) [inline src/Codacy/Coverage/Command/Clover.php]: I need. 

My PHPUnit has very long time execution. 
So I had already prepared clover.xml. In this situation I need:
* checkout for commit
* run PHPUnit
* run codacycoverage

Second reason - CI using. For working, I need go to build dir, and exec codacy in build dir. Sometimes make this seem like workaround. So if I have report in clover.xml and has commit, I simply want to send results, without any `cd` or `chroot`. 

It's option, not a argument.

**Comment by haffla** [comment]
@kirillkomlev very good work! I am OK with merging this after the minor issues (see my comments) are resolved.

**Pull request merged**